src/components/document/sort-work-view.tsx

```diff
@@ -33,11 +33,18 @@ export const SortWorkView: React.FC = observer(function SortWorkView() {
     persistentUI.setDocFilter(filter);
   };
 
+  const handlePrimarySortBySelection = (sort: string) => {
+    setPrimarySortBy(sort);
+    if (sort === secondarySortBy) {
+      setSecondarySortBy("None");
+    }
+  };
+
```

Comment:
This function and the rest of the changes in this file are not directly related to the story this PR is for. While adding tests for the other changes here, I discovered that the previous method of ensuring the primary and secondary sort options couldn't be set to the same thing wasn't quite right. While it would, for example, automatically change the selected secondary sort option from "Name" to "None" if the primary sort option was set to "Name", it wasn't leading to the HTML being fully updated. The secondary sort menu's Name option would still be marked as `selected`.

I had to implement a more thorough method to make it work correctly.